### PR TITLE
[vcpkg] fix cmake paths in tags (alternative)

### PR DESCRIPTION
**Pull request by Neumann-A**
alternative to #13710 (after discussion on slack or discord.... cannot remember anymore)

cc @ras0219-msft @ras0219

### CODE REVIEW

File: toolsrc/src/vcpkg/cmakevars.cpp [hunk 1]
```diff
@@ -88,12 +88,28 @@ namespace vcpkg::CMakeVars
...
+            {
+                if (cmake_current_list_dir_pos != 0)
+                {
+                    if (triplet_contents.compare(cmake_current_list_dir_pos - 1, std::string::npos, R"(\)"))
+                    {
+                        cmake_current_list_dir_pos += cmake_current_list_dir.length();
+                        continue;
+                    }
+                }
+                triplet_contents.replace(cmake_current_list_dir_pos, cmake_current_list_dir.length(), path_to_triplet);
```
Comment (strega-nil) [inline toolsrc/src/vcpkg/cmakevars.cpp:107]:
```suggestion
                triplet_contents.replace(cmake_current_list_dir_pos, cmake_current_list_dir.size(), path_to_triplet);
```

Comment (Neumann-A) [reply toolsrc/src/vcpkg/cmakevars.cpp:107]: because it is shorter?

Comment (strega-nil) [reply toolsrc/src/vcpkg/cmakevars.cpp:107]: It's the one we use as standard.

File: toolsrc/src/vcpkg/cmakevars.cpp [hunk 2]
```diff
@@ -88,12 +88,28 @@ namespace vcpkg::CMakeVars
...
+            const std::string cmake_current_list_dir{"${CMAKE_CURRENT_LIST_DIR}"};
+
+            auto cmake_current_list_dir_pos = triplet_contents.find(cmake_current_list_dir, 0);
+            while (cmake_current_list_dir_pos != std::string::npos)
+            {
+                if (cmake_current_list_dir_pos != 0)
+                {
+                    if (triplet_contents.compare(cmake_current_list_dir_pos - 1, std::string::npos, R"(\)"))
+                    {
+                        cmake_current_list_dir_pos += cmake_current_list_dir.length();
```
Comment (Neumann-A) [inline toolsrc/src/vcpkg/cmakevars.cpp:103]:
```suggestion
                        cmake_current_list_dir_pos += cmake_current_list_dir.size();
```

Comment (strega-nil) [reply toolsrc/src/vcpkg/cmakevars.cpp:103]: mmh, thanks

**Pull request closed without merging**